apps/system/js/base_module.js

```diff
@@ -111,6 +103,24 @@
   BaseModule.STATES = [];
 
   var SubmoduleMixin = {
+    // We could dynamically add and load new modules on demand.
+    loadWhenIdle: function(modules, time) {
+      return new Promise(function(resolve) {
+        this.constructor.SUB_MODULES =
+          this.constructor.SUB_MODULES.concat(modules);
+        var self = this;
+        var idleObserver = {
+          time: time || 10,
+          onidle: function() {
```

Comment:
`onidle` will invoke instantly if the phone has already idle for more than 10 sec when `addIdleObserver` is called. Are you sure this is what's intended? If so we can use IdleObserver directly.